src/DependencyInjection/PhpAmqpLibMessengerCompilerPass.php

```diff
@@ -11,42 +11,52 @@
 use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\DependencyInjection\Reference;
-use Symfony\Component\Messenger\MessageBusInterface;
-use Symfony\Component\Messenger\Transport\TransportInterface;
 
-use function str_starts_with;
+use function array_keys;
+use function sprintf;
 
 class PhpAmqpLibMessengerCompilerPass implements CompilerPassInterface
 {
     /** @throws InvalidArgumentException */
     #[Override]
     public function process(ContainerBuilder $container): void
     {
-        $serviceIds = $container->getServiceIds();
+        $transportRefs = [];
 
-        $transportServiceReferences = [];
+        foreach (array_keys($container->findTaggedServiceIds('messenger.receiver')) as $serviceId) {
+            $transportRefs[] = new Reference($serviceId);
+        }
 
-        foreach ($serviceIds as $serviceId) {
-            if (! str_starts_with($serviceId, 'messenger.transport.')) {
-                continue;
-            }
+        foreach (array_keys($container->findTaggedServiceIds('messenger.bus')) as $busId) {
+            $batchBusId = sprintf('%s.batch', $busId);
 
-            $definition = $container->getDefinition($serviceId);
+            $container->register($batchBusId, BatchMessageBus::class)
+                ->setArguments([
+                    new Reference($busId),
+                    $transportRefs,
+                ]);
 
-            if ($definition->getClass() !== TransportInterface::class) {
-                continue;
-            }
+            $container->registerAliasForArgument(
+                $batchBusId,
+                BatchMessageBusInterface::class,
+                $busId,
+            );
+        }
 
-            $transportServiceReferences[] = new Reference($serviceId);
+        if (! $container->hasAlias('messenger.default_bus')) {
+            return;
         }
 
-        $container->register(BatchMessageBus::class)
-            ->setClass(BatchMessageBus::class)
-            ->setArguments([
-                new Reference(MessageBusInterface::class),
-                $transportServiceReferences,
-            ]);
+        $defaultBusId = (string) $container->getAlias('messenger.default_bus');
+
+        $container->setAlias(
+            BatchMessageBus::class,
+            sprintf('%s.batch', $defaultBusId),
+        )->setPublic(true);
```

Comment:
I need to remove this. I can't figure out how to test services in this test without it being a public service https://github.com/jwage/phpamqplib-messenger/blob/main/tests/TransportFunctionalTest.php

I have this setup to be able to get services from the container in my tests, but I can't figure out how to replicate that here.